generate_sharktank.py

```diff
@@ -51,6 +54,59 @@ def save_torch_model(torch_model_list):
 
             model = None
             input = None
+            if model_type == "fx_imported":
+                from shark.examples.shark_inference.stable_diffusion.stable_args import (
+                    args,
+                )
+
+                args.use_tuned = False
+                args.import_mlir = True
+                args.use_tuned = False
+                args.local_tank_cache = WORKDIR
+                base_sd_versions = ["v1_4", "v2_1"]
+                model_variants = [
+                    "stablediffusion",
+                    "anythingv3",
+                    "analogdiffusion",
+                    "openjourney",
+                ]
+
+                precision_values = ["fp16"]
+                seq_lengths = [64, 77]
+                for model_variant in model_variants:
+                    args.variant = model_variant
+
+                    for base_sd_ver in base_sd_versions:
+                        # model variants not required for non base sd models
+                        if (
+                            base_sd_ver == "v1_4"
+                            and not model_variant == "stablediffusion"
+                        ):
+                            continue
+                        else:
+                            args.version = base_sd_ver
+
+                        for precision_value in precision_values:
+                            args.precision = precision_value
+                            for length in seq_lengths:
+                                model = mw.SharkifyStableDiffusionModel(
+                                    model_id="stabilityai/stable-diffusion-2-1-base",
+                                    custom_weights="",
+                                    precision=precision_value,
+                                    max_len=length,
+                                    width=512,
+                                    height=512,
+                                    use_base_vae=False,
+                                    debug=True,
+                                    sharktank_dir=WORKDIR,
+                                )
+                                args.max_length = length
+                                model_name = f"{args.variant}/{args.version}/{torch_model_name}/{args.precision}/length_{args.max_length}{args.use_tuned}"
```

Comment:
The model_name should have `width` and `height` params for future uploads.

Reply:
It should also have `batch_size`.

Reply:
this is actually dead code, my bad. I'm using the model name from model wrappers, so if we add batch size there it will propagate